### PR TITLE
Separate signal types from peerconnection

### PR DESCRIPTION
**Pull request by iislucas**
To be reviewed after: https://github.com/uProxy/uproxy-lib/pull/145 is merged and this pull request is updated to be against the latest head of dev (this pull request builds on that one).

There were a number of places (in uproxy-networking) where peerconnection was being imported just for signaling types; this was bad because it results in huge amounts of confused code in the browserified outputs (e.g. module-environment dependent code in a core env). So, I moved the signal types to their own file and now import only that when needed (done in: https://github.com/uProxy/uproxy-networking/pull/224)

Note: In merge; noticed that coverage was broken, so that has been fixed too. 

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/146)

### CODE REVIEW

**Comment by iislucas** [comment]
Thanks!

Renamed to `signals` I think I like that better too. 

Sadly your comment on JSON.stringify didn't have a location attached to it (maybe reviewable is better at that), but I'm guessing you're referring to log statements... I've removed JSON.stringify from those locations. I also noticed two potential bugs, fixed one, and added a TODO for the other. 

PTAL.

**Comment by jpevarnek** [comment]
Strange, it's showing up on the line for me...

I think you may have missed another comment I had above:
>General note: I would rename all the places you have used s as a variable name to message, I think that is a lot more descriptive (I might also do the same for some of the places the variable is still called signal actually)

**Comment by iislucas** [comment]
ooo, I did miss that, could call. fixed. PTAL.

**Comment by trevj** [comment]
I think this is the diff we're looking for:
https://github.com/uProxy/uproxy-lib/compare/iislucas-coverageRuleFixes...iislucas-separate-signal

File: src/samples/simple-freedom-chat/freedom-module.ts [hunk 1]
```diff
@@ -68,18 +68,18 @@ function makePeerConnection(name:string) {
...
-var b :PeerConnection<SignallingMessage> = makePeerConnection('B')
+var a :PeerConnection<signals.Message> = makePeerConnection('A');
+var b :PeerConnection<signals.Message> = makePeerConnection('B')
 
 // Connect the two signalling channels. Normally, these messages would be sent
 // over the internet.
-a.signalForPeerQueue.setSyncHandler((signal:SignallingMessage) => {
-  log.info('a: sending signal to b.');
-  b.handleSignalMessage(signal);
+a.signalForPeerQueue.setSyncHandler((message:signals.Message) => {
```
Comment (trevj) [inline src/samples/simple-freedom-chat/freedom-module.ts]: You've made this `signal:signals.Message` in some places, not in others. I prefer `signal` but we should be consistent.

**Comment by jpevarnek** [comment]
LGTM

File: src/samples/simple-freedom-chat/freedom-module.ts [hunk 2]
```diff
@@ -68,18 +68,18 @@ function makePeerConnection(name:string) {
...
+  log.info('a: sending signal message to b.');
+  b.handleSignalMessage(message);
 });
-b.signalForPeerQueue.setSyncHandler((signal:SignallingMessage) => {
-  log.info('b: sending signal to a.');
-  a.handleSignalMessage(signal);
+b.signalForPeerQueue.setSyncHandler((message:signals.Message) => {
+  log.info('b: sending signal message to a.');
```
Comment (trevj) [inline src/samples/simple-freedom-chat/freedom-module.ts]: "signal message" reads weird..."signalling message" would be better (or "signal" for short).

File: src/webrtc/peerconnection.ts [hunk 3]
```diff
@@ -78,7 +58,7 @@ export interface PeerConnection<TSignallingMessage> {
 
   // The |handleSignalMessage| function should be called with signalling
   // messages from the remote peer.
-  handleSignalMessage :(signal:TSignallingMessage) => void;
+  handleSignalMessage :(s:TSignallingMessage) => void;
```
Comment (trevj) [inline src/webrtc/peerconnection.ts]: any reason to go from `signal` -> `s`?

File: src/samples/simple-freedom-chat/freedom-module.ts (hunk 1 above)
Comment (iislucas) [inline src/samples/simple-freedom-chat/freedom-module.ts]: I think I'm now using `message::signals.Message` everywhere.

File: src/samples/simple-freedom-chat/freedom-module.ts (hunk 2 above)
Comment (iislucas) [inline src/samples/simple-freedom-chat/freedom-module.ts]: done.

File: src/webrtc/peerconnection.ts (hunk 3 above)
Comment (iislucas) [inline src/webrtc/peerconnection.ts]: done.

**Comment by iislucas** [comment]
@trevj: ping?

**Comment by trevj** [comment]
Sorry! Two easy comments, then :+1:.

---



---

<sup>**[src/webrtc/peerconnection.ts, line 392 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146#-JmZ6ST5FGptft7byA8T)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/48eafb8d8c4dc541fdc7b2b33b57636b372501eb/src/webrtc/peerconnection.ts#L392)):</sup>
This TODO does not read well :-)

---

<sup>**[src/webrtc/peerconnection.ts, line 447 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146#-JmZ5DB4JQIOeAWEtT7h)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/48eafb8d8c4dc541fdc7b2b33b57636b372501eb/src/webrtc/peerconnection.ts#L447)):</sup>
Indentation, semi-colons, and quotes need attention:

```typescript
 var descriptionString :string;
 try {
  descriptionString = JSON.stringify(description);
} catch (e) {
  descriptionString = '[unstringifiable object]';
}
this.closeWithError_('Failed to set remote description: ' +
    descriptionString + '; Error: ' + e.toString());
```

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146)

**Comment by iislucas** [comment]
---

<sup>**[src/webrtc/peerconnection.ts, line 392 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146#-JmZ6ST5FGptft7byA8T)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/48eafb8d8c4dc541fdc7b2b33b57636b372501eb/src/webrtc/peerconnection.ts#L392)):</sup>
oops, that comment was confused/wrong (only parse needs a try-catch). good catch. removed. 

---

<sup>**[src/webrtc/peerconnection.ts, line 447 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146#-JmZ5DB4JQIOeAWEtT7h)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/48eafb8d8c4dc541fdc7b2b33b57636b372501eb/src/webrtc/peerconnection.ts#L447)):</sup>
oops, same error here: stringify doesn't throw so this isn't needed. cleaned up.  

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146)

**Comment by iislucas** [comment]
Thanks for your diligent reviewing; that caught two confusions that nearly crept into the code! 

---



---



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146)

**Comment by jpevarnek** [comment]
---

<sup>**[src/webrtc/peerconnection.ts, line 444 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146#-JmZ5DB4JQIOeAWEtT7h)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/48eafb8d8c4dc541fdc7b2b33b57636b372501eb/src/webrtc/peerconnection.ts#L444)):</sup>
`stringify` will throw in some cases (at the very least if it is given a recursive data structure)

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146)

**Comment by iislucas** [comment]
---

<sup>**[src/webrtc/peerconnection.ts, line 444 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146#-JmZ5DB4JQIOeAWEtT7h)** ([raw file](https://github.com/uproxy/uproxy-lib/blob/48eafb8d8c4dc541fdc7b2b33b57636b372501eb/src/webrtc/peerconnection.ts#L444)):</sup>
good point. For these data structures, we are never going to get exceptions; so I'm fine without the runtime-check (we get compile time checking goodness here)

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/146)

**Pull request merged**